crates/runtime/src/logic.rs

```diff
@@ -426,4 +484,36 @@ impl VMHostFunctions<'_> {
         self.with_logic_mut(|logic| logic.registers.set(logic.limits, register_id, data))?;
         Ok(status)
     }
+
+    /// Generates a new UUID using v4 standard.
+    ///
+    /// This function generates a new UUID by calling the host. The UUID is
+    /// dependent on randomness, which is not available inside the guest
+    /// runtime. Therefore the guest needs to request this from the host.
+    ///
+    pub fn generate_uuid(&mut self, register_id: u64) -> VMLogicResult<()> {
+        let uuid = self.borrow_logic().generate_uuid();
+        self.with_logic_mut(|logic| {
```

Comment:
there's no fundamental reason to have a non-mut logic reference in this case, can be simplified to

```suggestion
        self.with_logic_mut(|logic| {
            let uuid = logic.generate_uuid();
```

Reply:
That's a fair observation 👍 Done! 🙂